Readable description for "InsufficientFundsForRent"

jdubpark opened the pull request:
Fixes #250

ngundotra commented on app/tx/[signature]/page-client.tsx:
```diff
@@ -1,34 +1,33 @@
-import { intoTransactionInstruction } from '@utils/tx';
-import { useClusterPath } from '@utils/url';
-import { BigNumber } from 'bignumber.js';
+import {Address} from '@components/common/Address';
+import {BalanceDelta} from '@components/common/BalanceDelta';
+import {ErrorCard} from '@components/common/ErrorCard';
+import {InfoTooltip} from '@components/common/InfoTooltip';
+import {LoadingCard} from '@components/common/LoadingCard';
+import {Signature} from '@components/common/Signature';
+import {Slot} from '@components/common/Slot';
+import {SolBalance} from '@components/common/SolBalance';
+import {TableCardBody} from '@components/common/TableCardBody';
+import {SignatureContext} from '@components/instruction/SignatureContext';
+import {InstructionsSection} from '@components/transaction/InstructionsSection';
+import {ProgramLogSection} from '@components/transaction/ProgramLogSection';
+import {TokenBalancesCard} from '@components/transaction/TokenBalancesCard';
+import {FetchStatus} from '@providers/cache';
+import {useCluster} from '@providers/cluster';
+import {useFetchTransactionStatus, useTransactionDetails, useTransactionStatus} from '@providers/transactions';
+import {useFetchTransactionDetails} from '@providers/transactions/parsed';
+import {SystemInstruction, SystemProgram, TransactionSignature} from '@solana/web3.js';
+import {Cluster, ClusterStatus} from '@utils/cluster';
+import {displayTimestamp} from '@utils/date';
+import {SignatureProps} from '@utils/index';
```
Can you run `yarn fmt` on your code & fix these no-op changes?

These changes do nothing to affect code, but affect readability of the PR

jdubpark replied:
Sorry, fixed the format. I don't know why my IDE is doing that (fixed by using vscode now)

ngundotra commented on app/utils/tx.ts:
```diff
@@ -558,3 +560,30 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
     };
 }
+
+export function intoTransactionErrorReason(info: TransactionStatusInfo, tx?: ParsedTransaction | Transaction): { errorReason: string, errorLink?: string } {
```
Can you rename this to `getTransactionErrorReason` & move this back to the page it came from?

It's not sufficiently general to be moved into `utils`

ngundotra commented on app/tx/[signature]/page-client.tsx:
```diff
@@ -48,6 +53,34 @@ type Props = Readonly<{
+        // ignore (return below)
+    }
+
+    return { errorReason: `Unknown Error: "${JSON.stringify(info.result.err)}"` }; // catch-all
```
Nit: Can you remove the comments from this function?

ngundotra commented on app/utils/tx.ts:
```diff
@@ -448,7 +448,7 @@ export const PROGRAM_INFO_BY_ID: { [address: string]: ProgramInfo } = {
 };
 
-export type LoaderName = typeof LOADER_IDS[keyof typeof LOADER_IDS];
+export type LoaderName = (typeof LOADER_IDS)[keyof typeof LOADER_IDS];
```
Nit: Can you remove the parantheses here?

steveluscher replied:
I think this is actually a new [Prettier feature](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACGATwzjwBkIBDAEzmwDlKBbUgXgOLggDNyB5AIIARAKIAlAPoBJIQGUA2gGs4hHhxJqyg0ZJmyAugG4QAGhAQMMAJbQAzslCVs2CAHcACk4T2UlADaulIT2ZgBG2JRgyjCyzHBkVlBwyNz+tnBhEVFwMRiRiQDmyDDYAK4ZqEyhcNS01GSUUAWllAVwAGI4TJQw1k3IIJSlMBCmIAAWMEx+AOrjVvC2eWBwst4LVgBuC4QDYLYhIInpuO4RBd0paRUAVrZosoV+cACKpRDwV37pZnnYJwMiCRbGBsFZLGMMGDYDMrNQYONkAAOAAMvxc6RmEQwAyhcBOm2SZgAju94GcLD5BrYALRJGo1MbYOCkqzMs6tS5IVLfCrpJhWYplPlPOACXpg0LDODuOgJJJfH4gWyit4fZLc65mGCUUKw+GIpAAJm1ESsfkKAGEIEwuahbABWMaldIAFV1Ph5Ss25SkUFosFkoPBMAE-tkRGeirgAF8Y0A). We should upgrade Prettier to the latest and make these changes.

ngundotra commented on app/tx/[signature]/page-client.tsx:
```diff
@@ -229,7 +263,13 @@ function StatusCard({ signature, autoRefresh }: SignatureProps & AutoRefreshProp
                         <td className="text-lg-end">
                             <h3 className="mb-0">
-                                <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                {errorLink ? (
+                                    <Link href={errorLink}>
+                                        <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                    </Link>
+                                ) : (
+                                    <span className={`badge bg-${statusClass}-soft`}>{errorReason}</span>
+                                )}
```
This actually works surprisingly well. No UI indicator mentioning it's a link, but still great.

ngundotra reviewed (changes requested):
Minor nits related to code hygiene, but otherwise looks great & think it's ready to ship!

cc @steveluscher

jdubpark commented:
Removed comments and removed the try-catch in `getTransactionErrorReason` for (future) other error catches.

Also, added `err.errorLink` validation to only link errors when a link is not undefined. Previously it led to errors linking to `/undefined`.

jdubpark reviewed (commented):
Fixed accordingly
